src/main/java/de/tum/in/www1/artemis/repository/fetchOptions/ProgrammingExerciseFetchOptions.java

```diff
@@ -0,0 +1,44 @@
+package de.tum.in.www1.artemis.repository.fetchOptions;
+
+import de.tum.in.www1.artemis.domain.Exercise_;
+import de.tum.in.www1.artemis.domain.ProgrammingExercise_;
+
+/**
+ * Fetch options for the {@link de.tum.in.www1.artemis.domain.ProgrammingExercise} entity.
+ * Each option specifies an entity or a collection of entities to fetch eagerly when using a dynamic fetching query.
+ */
+public enum ProgrammingExerciseFetchOptions implements FetchOptions {
```

Comment:
Should we move the fetch options into the repositories as inner classes?

Reply:
I talked with Stephan and have changed this.